pages/behaviors/manage.tsx

```diff
@@ -33,28 +32,53 @@ interface BehaviorAsProps {
     _id: string;
 }
 
-export default function manageBehaviorsPage({ behaviors }) {
+export default function manageBehaviorsPage() {
     // Verifies if user has the correct permissions
     const {allowed, role} = CheckUser(["Admin", "BCBA"])
     if(!allowed) return(<div>Redirecting...</div>);
 
-    const [behaviorList, setBehaviorList] = useState<BehaviorAsProps[]>(
-        behaviors.map((behavior: Behavior, idx: number) => ({
-            behaviorName: behavior.name,
-            description: behavior.description,
-            datatype: behavior.datatype,
-            id: idx + 1,
-            _id: behavior._id,
-        }))
-    );
+    // fetch the behavior data on the client side
+    const [behaviors, setBehaviors] = useState(null)
+    const [behaviorList, setBehaviorList] = useState([]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const response = await fetch(`/api/search/behavior`, { method: 'GET' });
+                if (response.ok) {
+                    const data = await response.json();
+                    setBehaviors(data);
+                } else {
+                    console.error('Failed to fetch data:', response.status, response.statusText);
+                }
+            } catch (error) {
+                console.error('Error fetching data:', error);
+            }
+        };
+        fetchData();
+    }, []);
+
+    useEffect(() => {
+        if (behaviors) {
+            const behaviorList = behaviors.map((behavior, idx) => ({
+                behaviorName: behavior.name,
+                description: behavior.description,
+                datatype: behavior.datatype,
+                id: idx + 1,
+                _id: behavior.id,
+            }));
+            setBehaviorList(behaviorList);
+        }
+    }, [behaviors]);

```

Comment:
I can't quite seem to grasp the need for this second useEffect. 

Why do we need to map into a second state `behaviorList`? Why can't we use just only one piece of state? If we change this we would also need to change the way we reference data below in the template. We might also need to remove the whole extra `_id` field that we have, I can't understand why we need it. Might be a lack of context on my end though.


Secondly, if we were to map to a second object, it would make more sense to just do it in the first useEffect. Also currently we don't even use `behaviors` so why have it? If we wanted to keep a different mapped version, why not just only keep track of that? Again though I think that we don't actually need to map it here (we might need to do some extra changes like getting rid of the extra _id field).

@Roy0531 I think we should discuss this over a call.

Reply:
I think that we should change this to just use the same model that comes from prisma. Might have to do a few extra changes but I think it makes more sense. 

I might be missing something though.